clip/eval.py

```diff
@@ -0,0 +1,55 @@
+# Mirror of the Linear Probe Evaluation Script
+# from the official CLIP Repository.
+import mlx.core as mx
+import numpy as np
+from image_processor import CLIPImageProcessor
+from mlx.data.datasets import load_cifar10
+from model import CLIPModel
+from PIL import Image
+from sklearn.linear_model import LogisticRegression
+
+
+def get_cifar10(batch_size, root=None):
+    tr = load_cifar10(root=root)
+    tr_iter = tr.to_stream().batch(batch_size)
+
+    test = load_cifar10(root=root, train=False)
+    test_iter = test.to_stream().batch(batch_size)
+
+    return tr_iter, test_iter
+
+
+def get_features(model, image_proc, iter):
+    all_features = []
+    all_labels = []
+
+    for _, batch in enumerate(iter):
+        image, label = batch["image"], batch["label"]
+        x = image_proc([Image.fromarray(im) for im in image])
+        y = mx.array(label)
+
+        image_embeds = model.get_image_features(x)
+        all_features.append(image_embeds)
+        all_labels.append(y)
```

Comment:
You need an `mx.eval(image_embeds)` at some point otherwise you just create a huge graph for the GPU to compute at the same time which leads to memory problems.